### ISSUE TITLE
Default language fallback for comments if none present (Comments vanish?)

### ISSUE THREAD

**Issue by jancborchardt** [nextcloud/appstore#408]
I left a comment at the Mail app with my rating: https://apps.nextcloud.com/apps/mail and there were also some other comments before.

Now there are none. Is this maybe due to a new version being released? If so, the comments should only indicate that they were made before the recent version. But definitely not deleted.

@nextcloud/appstore

**Comment by BernhardPosselt** [comment]
Change your language, comments are only displayed in the language you've posted

**Comment by jancborchardt** [comment]
Maybe if there are no comments in your language then English should be shown by default? Just to populate it a bit in the beginning since most people will comment in english.

**Comment by BernhardPosselt** [comment]
That would probably be very confusing. If you post a comment in your language then one comment would be present and all English comments are gone. That's why you've filled your issue btw

**Comment by jancborchardt** [comment]
No, I opened the issue because I posted the comment in english and I had my browser in english. Then I viewed it with a different browser set to french.

So basically everyone who has their browser set to a different language than English will think the appstore is basically devoid of other people in the beginning. That’s not a good first-run-experience.

As a different solution, how about for the first few weeks/months we just show _all_ comments regardless of the language? Most will be english anyway.

**Comment by BernhardPosselt** [comment]
Providing a drop down to choose the language would solve this issue then I guess. The current language is preselected in the drop down and you can switch to a different language. 

BTW: Not many people use browsers in two different languages and logging in/choosing the language gets rid of that issue ;D

**Comment by jancborchardt** [comment]
> Providing a drop down to choose the language would solve this issue then I guess.

No it wouldn’t.

> BTW: Not many people use browsers in two different languages and logging in/choosing the language gets rid of that issue ;D

That’s not my point. The point is that _any_ person using any other language than English will see no comments at all for any app. Because most if not all comments will be made in English.
So it seems like a dead appstore, which is not a good experience. So let’s show them all, at least for the first few weeks/months.

**Comment by adsworth** [comment]
I think I'd currently agree with @jancborchardt that it's best to show other language comments rather than no comments.

How about this: If there are no comments in the users language we show a message "There are no comments in your language. Displayed are the comments in other languages" (or something to that effect) and then display the comments from all other languages.

This is a little heavier on the DB since we need at least two queries, but from a UX I think it's the best solution.

**Comment by Spartachetto** [comment]
I'll put here my user experience just as a contribute to the decision.

As soon as I read this issue, I changed the default language of my browser to English just to see the comments in the appstore. 
I had no idea they were there and was too curious to look at them.

As you can imagine I share the point made by @jancborchardt !

**Comment by BernhardPosselt** [comment]
@adsworth right, if no ratings are present in a language we should notify the user about that. @Spartachetto your issue can be handled entirely by providing a drop down to toggle between rating languages. 

We should probably also let the user file a rating in a different language so that the process where ratings go is a bit more transparent. That would also get rid of the confusion that I described further up https://github.com/nextcloud/appstore/issues/408#issuecomment-268822606 so we could default to English comments if no comments exist in that language.

TL;DR:
* Add a dropdown to select a language when posting a rating. Defaults to the user's browser language or profile language if logged in.
* Add a dropdown to select a language for displaying ratings. Defaults to the user's browser language or profile language if logged in. If no ratings are present it defaults to English. Changing the dropdown to a different language will reload the comments in that language (shouldn't be hard, they are currently loaded over AJAX anyways).

**Comment by jancborchardt** [comment]
> Your issue can be handled entirely by providing a drop down to toggle between rating languages.

@BernhardPosselt no. People will not change that language. Please just make it show the comments present in English, like @adsworth suggested.

If you don’t want to implement it then don’t just find silly arguments to circumvent it. This is currently a really shitty UX and we should fix it.

**Comment by BernhardPosselt** [comment]
@jancborchardt did you read the entire post? Because it's basically what you've suggested plus a way to quickly view and post comments in other languages?

**Comment by jancborchardt** [comment]
@BernhardPosselt BIG FAT SORRY! :) Thought I read the entire post but apparently didn’t. Yes, what you described is a great solution.

**Comment by adsworth** [comment]
If you change your language in Account /Change language the change is only saved in the users session not in a profile. That might be a little confusing when the sessions resets.

**Comment by BernhardPosselt** [comment]
@adsworth good to know, is there a way to make it permanent?

**Comment by adsworth** [comment]
@BernhardPosselt we'd have to create a Profile model. :)
See https://docs.djangoproject.com/en/1.10/topics/auth/customizing/#extending-the-existing-user-model 
the second paragraph onwards.

**Comment by BernhardPosselt** [comment]
Yeah, ofc :D

What I'm more interested in: do we need to add custom middleware to set the language from the user profile?

**Comment by adsworth** [comment]
Any objections to closing this issue?
and should we really save the users language to the DB, instead of just the session? If so then I'd open a new issue for that.

**Comment by BernhardPosselt** [comment]
This issue is basically superseded by two new issues, so I guess you could close it. 

> and should we really save the users language to the DB, instead of just the session

Yes.

**Comment by jancborchardt** [comment]
So … is it fixed?

**Comment by adsworth** [comment]
It should be. @BernhardPosselt did we deploy a new version after merging #414?

**Comment by BernhardPosselt** [comment]
Not yet. This issue is tracked in two follow up issues BTW. Will try to review things this weekend

**Comment by jancborchardt** [comment]
Cool, thanks for the work on it! :)
